edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleIdentityLifecycleManager.cs

```diff
@@ -77,13 +77,15 @@ public ModuleIdentityLifecycleManager(IIdentityManager identityManager, ModuleId
             IEnumerable<Task<Identity>> updateTasks = updateIdentities.Select(i => this.identityManager.UpdateIdentityAsync(i.ModuleId, i.GenerationId, i.ManagedBy));
             Identity[] upsertedIdentities = await Task.WhenAll(createTasks.Concat(updateTasks));
 
-            List<IModuleIdentity> moduleIdentities = upsertedIdentities.Select(m => this.GetModuleIdentity(m)).ToList();
-            // Add the module identity for Edge Agent in the returned dictionary
-            // because is was excluded from the update call.
-            if (identities.ContainsKey(Constants.EdgeAgentModuleIdentityName))
-            {
-                moduleIdentities.Add(this.GetModuleIdentity(identities[Constants.EdgeAgentModuleIdentityName]));
-            }
+            List<IModuleIdentity> moduleIdentities;
+
+            moduleIdentities = upsertedIdentities.Select(m => this.GetModuleIdentity(m)).ToList();
+
+            // Add back the unchanged identities (including Edge Agent).
+            var removedIdentityList = removeIdentities.ToList();
+            var upsertedIdentityList = moduleIdentities.Select(i => i.ModuleId).ToList();
+            var unchangedIdentities = identities.Where(i => !removedIdentityList.Contains(i.Key) && !upsertedIdentityList.Contains(i.Key));
```

Comment:
removedIdentities does not contain identities that are not managed by edge, I think those should not be included as well. You can filter by removedModuleNames